### DISCUSSION TITLE
to_image causes 'map' object is not subscriptable in Image.py, line 1203

### DISCUSSION THREAD

**Discussion by tye-shutty** [jsvine/pdfplumber#637] [Q&A] [answered]
Hi,

I have this code working on one computer, but not the other:

```
import pdfplumber

pdf_obj = pdfplumber.open('existing1.pdf')
page = pdf_obj.pages[0]
images_in_page = page.images
page_height = page.height
image = images_in_page[3]
file_name = "test.png"
image_bbox = (image['x0'], page_height - image['y1'],
              image['x1'], page_height - image['y0'])
try:
  cropped_page = page.crop(image_bbox)
  image_obj = cropped_page.to_image(resolution=400)
  image_obj.save(file_name, format="PNG")
except ValueError:
  print('image outside of crop area')
```

traceback:

```
Traceback (most recent call last):
  File "/home/mo/codebase/pdf-image-locator/image_coords_demo.py", line 13, in <module>
    image_obj = cropped_page.to_image(resolution=400)
  File "/home/mo/.local/lib/python3.9/site-packages/pdfplumber/page.py", line 296, in to_image
    return PageImage(self, **kwargs)
  File "/home/mo/.local/lib/python3.9/site-packages/pdfplumber/display.py", line 74, in __init__
    self.original = self.original.crop(map(int, cropbox))
  File "/home/mo/.local/lib/python3.9/site-packages/PIL/Image.py", line 1205, in crop
    if box[2] < box[0]:
TypeError: 'map' object is not subscriptable

```

Working computer (Ubuntu 18.04.6 LTS) has python 3.7.6, ImageMagick 7.1.0-19
Failing computer (Ubuntu 21.10) has python 3.7 and python 3.9.7 (both fail), ImageMagick 7.1.0-2
edit: both use pdfplumber 0.6.0

[existing1.pdf](https://github.com/jsvine/pdfplumber/files/8426981/existing1.pdf)

**Comment by jsvine** [comment, marked as answer]
Hi @tye-shutty, and thanks for posting. Looking at the traceback, and investigating further, this seems to have something to do with newer versions of [`PIL/pillow`](https://pillow.readthedocs.io/en/stable/). As defined in its `requirements.txt`, `pdfplumber` requires `Pillow>=8.4`. The code works fine for me with `8.4` (and `9.0`) but falis (with the same error) when I upgrade to the latest version, `9.1`, released just 10 days ago. (Specifically, it appears to stem from the addition of [these lines checking for negative crops](https://github.com/python-pillow/Pillow/blame/5d070222d21138d2ead002fd33fdf5adcb708941/src/PIL/Image.py#L1203-L1204).)

If you'd like to check for your particular case, can you add this line to your script and rerun on your two machines:

```python
import PIL
print(PIL.__version__)
```

If that's indeed the case for you, the immediate fix would be to run ``pip install pillow==9.0`` in your relevant environment. In the long run, I'll aim to release a fix to the relevant code in `pdfplumber`.
